ndcube/tests/test_ndcube.py

```diff
@@ -1453,6 +1453,56 @@ def test_cube_arithmetic_multiply(ndcube_2d_ln_lt_units, value):
     # TODO: test that uncertainties scale correctly
 
 
+@pytest.mark.parametrize(("ndc", "value"),
+                        [
+                            ("ndcube_2d_ln_lt_no_unit_no_unc_no_mask_2", NDData(np.ones((2, 3)),
+                                                                                wcs=None) # neither of the two has uncertainty or mask or unit.
+                            ),
+                            ("ndcube_2d_ln_lt_no_unit_no_unc_no_mask_2", NDData(np.ones((2, 3)),
+                                                                                wcs=None,
+                                                                                uncertainty=StdDevUncertainty(np.ones((2, 3))*0.1),
+                                                                                mask=np.ones((2, 3), dtype=bool),
+                                                                                unit=u.ct) # ndc has no mask no uncertainty no unit, but nddata has all.
+                            ),
+                            ("ndcube_2d_ln_lt_unit_unc_mask", NDData(np.ones((2, 3)),
+                                                                                wcs=None,
+                                                                                uncertainty=StdDevUncertainty(np.ones((2, 3))*0.1),
+                                                                                mask=np.ones((2, 3), dtype=bool),
+                                                                                unit=u.ct) # both of them have uncertainty and mask and unit.
+                            )
+                        ],
+                        indirect=("ndc",))
+def test_cube_arithmetic_multiply_ndcube_nddata(ndc, value):
```

Comment:
It would probably be better to simplify this test by including the `expected_cube` in the parameterization.  Then the test only has to calculate the `output_cube` and compare it to the `expected_cube`.